Add a document 'How to Write Tests Using FileCheck'

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#19005 Add a document 'How to Write Tests Using FileCheck'**

Differential Revision: [D14826845](https://our.internmc.facebook.com/intern/diff/D14826845)

Commit:
Add a document 'How to Write Tests Using FileCheck'

eellison commented on test/HowToWriteTestsUsingFileCheck.md:
```diff
@@ -0,0 +1,100 @@
+      """
+    parsed = parse_ir(input_str)
+    optimized = run_cse(parsed)
+    FileCheck().run(input_str, optimized)
```
Could you add a comment about using the builder apI somewhere as well ? 
`FileCheck().check("aten::mul")`

eellison reviewed (approved):
Looks great! thanks for doing this

Commit:
Update on "Add a document 'How to Write Tests Using FileCheck'"

Add a document 'How to Write Tests Using FileCheck'

gh-metadata: pytorch pytorch 19005 gh/ZolotukhinM/7/head
